app/src/main/java/co/nimblehq/compose/crypto/ui/navigation/AppNavigation.kt

```diff
@@ -1,16 +1,22 @@
 package co.nimblehq.compose.crypto.ui.navigation
 
 import androidx.compose.runtime.Composable
+import androidx.compose.ui.res.stringResource
 import androidx.navigation.*
 import androidx.navigation.compose.*
+import co.nimblehq.compose.crypto.R
+import co.nimblehq.compose.crypto.ui.common.AppDialogPopUp
 import co.nimblehq.compose.crypto.ui.screens.detail.DetailScreen
 import co.nimblehq.compose.crypto.ui.screens.home.HomeScreen
 
 @Composable
 fun AppNavigation(
-    navController: NavHostController = rememberNavController(),
-    startDestination: String = AppDestination.Home.destination
+    navController: NavHostController,
+    startDestination: String = AppDestination.Home.destination,
+    onCallBackChange: (() -> Unit) -> Unit,
+    globalDialogCallback: () -> Unit
```

Comment:
@kaungkhantsoe Because of the actions for the application scope so I'd suggest avoiding single terms such as `onCallBackChange` or `globalDialogCallback`. Please specify the purpose instead, for me at first look, it's confused about the purpose of `onCallBackChange` and `globalDialogCallback`.


```suggestion
    onNetworkReconnectedCallback: (() -> Unit) -> Unit,
    onGlobalDialogActionButtonClicked: () -> Unit
```